### PR TITLE
Link to the calendar

### LINKED ISSUE #216: [Feature] Include room schedules
**Is your feature request related to a problem? Please describe.**
The roomfinder made by the department of physics includes a calendar with the schedule for a room which I find quite useful: https://www.ph.tum.de/about/visit/roomfinder/?room=5510.02.001&language=de

**Describe the solution you'd like**
Perhaps we could add a similar feature to NavigaTUM.

**Describe alternatives you've considered**
Look up the schedule on tumonline. Maybe add a link to e.g. https://campus.tum.de/tumonline/ee/ui/ca2/app/desktop/#/pl/ui/$ctx/wbKalender.wbRessource?$ctx=design=ca2;header=max;lang=de&pResNr=12543 ?

### PR DESCRIPTION
**Pull request by CommanderStorm**
This PR implements the link to the TUMOnline Calendar from #216 
image

Open questions:
- is the place for the data extraction (`structure`) the right place?
- is the api change the best option, or should this be a `prop`?
- is [the chosen icon](https://icons.getbootstrap.com/icons/calendar4-week/) a good solution?

Resolves  #216

### CODE REVIEW

File: webclient/src/views/view/view-view.inc [hunk 1]
```diff
@@ -94,6 +94,16 @@
...
       <div class="column col-auto col-ml-auto">
+        <a
+          class="btn btn-link btn-action btn-sm"
+          v-if="view_data.calendar_url"
+          v-bind:href="view_data.calendar_url"
+        >
+          <svg xmlns="http://www.w3.org/2000/svg" height="14" width="14" fill="currentColor" viewBox="0 0 16 16" style="margin-bottom: -1px">
+            <path d="M3.5 0a.5.5 0 0 1 .5.5V1h8V.5a.5.5 0 0 1 1 0V1h1a2 2 0 0 1 2 2v11a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2V3a2 2 0 0 1 2-2h1V.5a.5.5 0 0 1 .5-.5zM2 2a1 1 0 0 0-1 1v1h14V3a1 1 0 0 0-1-1H2zm13 3H1v9a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1V5z"/>
+            <path d="M11 7.5a.5.5 0 0 1 .5-.5h1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-1a.5.5 0 0 1-.5-.5v-1zm-3 0a.5.5 0 0 1 .5-.5h1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-1a.5.5 0 0 1-.5-.5v-1zm-2 3a.5.5 0 0 1 .5-.5h1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-1a.5.5 0 0 1-.5-.5v-1zm-3 0a.5.5 0 0 1 .5-.5h1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-1a.5.5 0 0 1-.5-.5v-1z"/>
+          </svg>
```
Comment (octycs) [inline webclient/src/views/view/view-view.inc:105]:
```suggestion
          <svg 
            xmlns="http://www.w3.org/2000/svg"
            width="14"
            height="14"
            viewBox="0 0 16 16"
            fill="currentColor" 
            style="margin-bottom: -2px"
          >
            <path d="M4.571 0A1.143 1.143 0 0 0 3.43 1.143H2.286A2.306 2.306 0 0 0 0 3.429v10.285A2.306 2.306 0 0 0 2.286 16h11.428A2.306 2.306 0 0 0 16 13.714V3.43a2.306 2.306 0 0 0-2.286-2.286h-1.143A1.143 1.143 0 0 0 11.43 0a1.143 1.143 0 0 0-1.143 1.143H5.714A1.143 1.143 0 0 0 4.571 0zM2.286 5.714h11.428v8H2.286v-8z"/>
            <path d="M6.857 6.857v2.286h2.286V6.857H6.857zm3.429 0v2.286h2.285V6.857h-2.285zm-6.857 3.429v2.285h2.285v-2.285H3.43zm3.428 0v2.285h2.286v-2.285H6.857z"/>
          </svg>
```

image

File: webclient/src/views/view/view-view.inc [hunk 2]
```diff
@@ -94,6 +94,16 @@
         <span>{{ view_data.type_common_name }}</span>
       </div>
       <div class="column col-auto col-ml-auto">
+        <a
+          class="btn btn-link btn-action btn-sm"
+          v-if="view_data.calendar_url"
+          v-bind:href="view_data.calendar_url"
```
Comment (octycs) [inline webclient/src/views/view/view-view.inc:100]:
```suggestion
          v-bind:href="view_data.calendar_url"
          target="_blank"
```
Would you open in a new tab?

**Review by octycs — APPROVED** [review]
Sorry for the long delay. I've created a new icon that I think matches the style of the others a bit more. And I'm slightly in preference of a new prop instead (it's a bit cleaner in my opinion), but I'm not opposed to do it this way.

**Comment by octycs** [comment]
I don't know how to best put it here without creating conflict: We can also add a Tooltip by adding to the `<a>` a
```html
<a
    ...
    title="${{ _.view_view.header.calendar }}$"

```
 and in `i18n-view.yaml` add:
```yaml
view_view:
  // ...
  header:
    calendar: { de: "Kalender öffnen", en: "Open calendar" }
```

**Comment by CommanderStorm** [comment]
> And I'm slightly in preference of a new prop instead (it's a bit cleaner in my opinion), but I'm not opposed to do it this way.

Read: Frank, please change this, or
Read: This is fine as is

**Comment by octycs** [comment]
> Read: Frank, please change this, or
> Read: This is fine as is

I'm sorry I wasn't clear what I meant. I have no strong opinion on this. I would probably have chosen a prop, and if you think this is fine for you as well, I'd prefer it this way. But if you prefer it not being a prop, I won't be opposed.

**Comment by CommanderStorm** [comment]
> I have no strong opinion on this.

I also dont, the reason why I cose this place is, because props states, that it is `Data for the info-card table` and this is not in this table..

**Comment by CommanderStorm** [comment]
I have implementet the change in https://github.com/TUM-Dev/NavigaTUM/pull/221/commits/6292a219cb8fcb7f1e748054fd9317243433ab70

**Pull request merged**